// @wordpress/url/build-module/index.js
import { isURL } from "./is-url";
import { isEmail } from "./is-email";
import { isPhoneNumber } from "./is-phone-number";
import { getProtocol } from "./get-protocol";
import { isValidProtocol } from "./is-valid-protocol";
import { getAuthority } from "./get-authority";
import { isValidAuthority } from "./is-valid-authority";
import { getPath } from "./get-path";
import { isValidPath } from "./is-valid-path";
import { getQueryString } from "./get-query-string";
import { buildQueryString } from "./build-query-string";
import { isValidQueryString } from "./is-valid-query-string";
import { getPathAndQueryString } from "./get-path-and-query-string";
import { getFragment } from "./get-fragment";
import { isValidFragment } from "./is-valid-fragment";
import { addQueryArgs } from "./add-query-args";
import { getQueryArg } from "./get-query-arg";
import { getQueryArgs } from "./get-query-args";
import { hasQueryArg } from "./has-query-arg";
import { removeQueryArgs } from "./remove-query-args";
import { prependHTTP } from "./prepend-http";
import { safeDecodeURI } from "./safe-decode-uri";
import { safeDecodeURIComponent } from "./safe-decode-uri-component";
import { filterURLForDisplay } from "./filter-url-for-display";
import { cleanForSlug } from "./clean-for-slug";
import { getFilename } from "./get-filename";
import { normalizePath } from "./normalize-path";
import { prependHTTPS } from "./prepend-https";
export {
  addQueryArgs,
  buildQueryString,
  cleanForSlug,
  filterURLForDisplay,
  getAuthority,
  getFilename,
  getFragment,
  getPath,
  getPathAndQueryString,
  getProtocol,
  getQueryArg,
  getQueryArgs,
  getQueryString,
  hasQueryArg,
  isEmail,
  isPhoneNumber,
  isURL,
  isValidAuthority,
  isValidFragment,
  isValidPath,
  isValidProtocol,
  isValidQueryString,
  normalizePath,
  prependHTTP,
  prependHTTPS,
  removeQueryArgs,
  safeDecodeURI,
  safeDecodeURIComponent
};
